src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -54,31 +54,47 @@ const getInitialReportScreenParams = (reports, ignoreDefaultRooms, policies) =>
     return {reportID: String(reportID)};
 };
 
-const MainDrawerNavigator = (props) => {
-    const initialParams = getInitialReportScreenParams(props.reports, !Permissions.canUseDefaultRooms(props.betas), props.policies);
+class MainDrawerNavigator extends Component {
+    constructor(props) {
+        super(props);
+        this.initialParams = getInitialReportScreenParams(props.reports, !Permissions.canUseDefaultRooms(props.betas), props.policies);
+    }
+
+    shouldComponentUpdate(nextProps) {
+        const initialNextParams = getInitialReportScreenParams(nextProps.reports, !Permissions.canUseDefaultRooms(nextProps.betas), nextProps.policies);
+        if (this.initialParams.reportID === initialNextParams.reportID) {
+            return false;
+        }
 
-    // Wait until reports are fetched and there is a reportID in initialParams
-    if (!initialParams.reportID) {
-        return <FullScreenLoadingIndicator logDetail={{name: 'Main Drawer Loader', initialParams}} />;
+        // Probably better to be placed inside shouldComponentUpdate but we already calculated next params so assigning immediately
```

Comment:
This comment seems to imply that there is some better place to put this, but I'm not really following it. I think it can just be removed.